src/token/feature-ext-parser.js

```diff
@@ -0,0 +1,79 @@
+const async = require('async');
+
+const FedAuthInfoID = {
+  STSURL: 0x01,
+  SPN: 0x02
+};
+
+const TERMINATOR = 0xFF;
+
+const infoData = {
+  spnLen: undefined,
+  stsurlLen: undefined
+};
+
+var readFedAuthInfoOpt = (parser, callback) => {
+  parser.readUInt8((fedauthInfoID) => {
+    parser.readUInt32LE((fedAuthInfoDataLen) => {
+      parser.readUInt32LE((fedAuthInfoDataOffset) => {
+
+        switch (fedauthInfoID) {
+          case FedAuthInfoID.SPN:
+            infoData.spnLen = fedAuthInfoDataLen ;
+            break;
+          case FedAuthInfoID.STSURL:
+            infoData.stsurlLen = fedAuthInfoDataLen ;
+            break;
+            // ignoring unknown fedauthinfo options
+          default:
+            break;
+        }
+        if (infoData.spnLen && infoData.stsurlLen) {
+          parser.readBuffer(infoData.spnLen, (spn) => {
+            parser.readBuffer(infoData.stsurlLen, (stsurl) => {
+              callback({
+                'name': 'FEDAUTHINFO',
+                'event': 'fedAuthInfo',
+                'fedAuthInfoData': {
+                  stsurl: stsurl.toString().replace(/\0/g, ''),
+                  spn: spn.toString().replace(/\0/g, '')
+                }
+              });
+            });
+          });
+        }
+      });
+    });
+  });
+};
+
+module.exports.fedAuthInfoParser = fedAuthInfoParser;
+function fedAuthInfoParser(parser, colMetadata, options, callback) {
+  parser.readUInt32LE((tokenLength) => {
+    parser.readUInt32LE((countOfInfoIDs) => {
+      async.times(countOfInfoIDs, () => {
+        readFedAuthInfoOpt(parser, callback);
+      });
+    });
+  });
+}
+
+module.exports.featureExtAckParser = featureExtAckParser;
+function featureExtAckParser(parser, colMetadata, options, callback) {
```

Comment:
What do you think about breaking this up into two separate files, one for parsing `FEDAUTHINFO` and the other for `FEATUREEXTACK`?